### PR TITLE
[feature] add a Circuit method for Reset

### PR DESCRIPTION
**Pull request by ss2165**
Commonly used enough that having to switch to add_gate confuses people.

### CODE REVIEW

**Review by cqc-melf — APPROVED** [review]
Are there any examples in which we want to add this?

**Comment by cqc-melf** [comment]
You need to update the version in the binders as well: https://github.com/CQCL/tket/blob/feature/reset-method/pytket/conanfile.py#L35

But I am not sure if that solves the problem.

If you need help I could take a more detailed look later today

File: pytket/pytket/_tket/circuit.pyi [hunk 1]
```diff
@@ -471,20 +471,6 @@ class Circuit:
         :return: the new :py:class:`Circuit`
         """
     @typing.overload
```
Comment (cqc-melf) [inline pytket/pytket/_tket/circuit.pyi:473]: The stub generation removes this, will look into this again tomorrow.

Comment (ss2165) [reply pytket/pytket/_tket/circuit.pyi:473]: yes I think I mistakenly put those in manually, should be taken out and generated I think?

Comment (cqc-melf) [reply pytket/pytket/_tket/circuit.pyi]: Should be working now

**Review by cqc-melf — APPROVED** [review]

File: pytket/pytket/_tket/circuit.pyi (hunk 1 above)
Comment (ss2165) [reply pytket/pytket/_tket/circuit.pyi]: thank you!

**Pull request merged**